src/tablestore.rs

```diff
@@ -361,19 +390,22 @@ impl TableStore {
             object_store: self.object_store.clone(),
```

Comment:
Mind removing `read_blocks_using_index` and replacing it with `read_blocks` (above) instead? (This is what the TODO above is mentioning.)

Reply:
Hmm, yea. Let's leave it for now.

I guess this comment was assuming the `read_index` call in the `SstIterator` would either not be needed, or would warm the cache, so the `read_blocks` call would index would always be in cache when `read_blocks` was called. But there is a risk that the index could get dropped from the cache in-between. We'd have to re-fetch it if we depended on the cache only, rather than using `SstIterator`'s saved index.

@rodesai Am I missing something from this comment?

Reply:
I put this fn in because I didn't want to make perf worse when moving the index out of the SST info. I was thinking once we had a cache, we could just rely on the index being cached and drop `read_blocks_using_index` - hence the TODO. At the same time, we probably don't want to cache the index in the main cache when iterating over SSTs for compaction as they can be quite large (100s of MBs of index for large sorted runs). And we also don't want to read the index for every block-range read as it can be quite large (multiple MBs). So I'm in favor of just dropping the TODO.